### ISSUE TITLE
Cases analysis on unknown type

### ISSUE THREAD

**Issue by cpitclaudel** [ocaml/merlin#1029]
TL;DR: Could `merlin-destruct` be enhanced to prompt for a type when it doesn't know the type of the current context?

Background: This is a feature from Coq that I miss when using Merlin.  In PG+company-coq I can use a keybinding to start a `match` statement, and Emacs prompts me for the type to destruct (so if I type `option` I get `Some x` and `None`, for example).

This is useful because I often don't have a type annotation on the argument that I'm destructing, so Merlin doesn't know what type it has (I often expect the type of the argument to be inferred from the cases, but to create the cases in the first place I need to a way to give Merlin the type).  This would also be useful for creating `function` blocks (for these, I usually add a parameter to my function, then add a type annotation, then use `merlin-destruct` on the name of the parameter, then remove the parameter and the annotation).

**Comment by trefis** [comment]
> Could merlin-destruct be enhanced to prompt for a type when it doesn't know the type of the current context?

Yes.

I've been wanting this for a while, but haven't had time to do it yet.
